modules/nvidia_plugin/src/cuda_compiled_model.cpp

```diff
@@ -0,0 +1,386 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <ie_metric_helpers.hpp>
+// ^^ must come before ie_plugin_config.hpp, which is indirectly included by
+// cuda_executable_network.hpp
+
+#include <fmt/format.h>
+
+#include <memory_manager/cuda_memory_manager.hpp>
+#include <ops/nop_op.hpp>
+#include <ops/subgraph.hpp>
+#include <threading/ie_executor_manager.hpp>
+#include <utility>
+
+#include "cuda_compiled_model.hpp"
+#include "cuda_itt.hpp"
+#include "cuda_operation_registry.hpp"
+#include "cuda_plugin.hpp"
+#include "memory_manager/cuda_immutable_memory_block_builder.hpp"
+#include "memory_manager/cuda_memory_manager.hpp"
+#include "memory_manager/model/cuda_memory_model_builder.hpp"
+#include "nvidia/nvidia_config.hpp"
+#include "nvidia/properties.hpp"
+
+#include "ops/parameter.hpp"
+#include "ops/result.hpp"
+#include "transformations/utils/utils.hpp"
+#include "transformer/cuda_graph_transformer.hpp"
+
+#include "openvino/runtime/exec_model_info.hpp"
+#include "openvino/runtime/iplugin.hpp"
+#include "openvino/pass/serialize.hpp"
+
+namespace {
+static constexpr const char* nv_task_executor_name = "NvidiaCallbackExecutor";
+static constexpr const char* nv_callback_executor_name = "NvidiaCallbackExecutor";
+}  // namespace
+
+namespace ov {
+namespace nvidia_gpu {
+
+using Time = std::chrono::steady_clock;
+
+CompiledModel::CompiledModel(const std::shared_ptr<const ov::Model>& model,
+                             const Configuration& cfg,
+                             const std::shared_ptr<ov::threading::ITaskExecutor>& wait_executor,
+                             const std::shared_ptr<const ov::IPlugin>& plugin,
+                             bool loaded_from_cache)
+    : ov::ICompiledModel(model, plugin, nullptr, nullptr),
+      config_(std::move(cfg)),
+      cuda_stream_executor_(std::move(wait_executor)),
+      loaded_from_cache_(loaded_from_cache) {
+    try {
+        compile_model(model);
+        init_executor();  // creates thread-based executor using for async requests
+        benchmark_optimal_number_of_requests();
+    } catch (const ov::Exception&) {
+        throw;
+    } catch (const InferenceEngine::Exception&) {
+        throw;
+    } catch (const std::exception& e) {
+        throwIEException(fmt::format("Standard exception from compilation library: {}", e.what()));
```

Comment:
@ilya-lavrenov What's the point of having such kind of exceptions handling? I see the similar code in template plugin

Reply:
We wrap exceptions to ov type.